### PR TITLE
 Additional gather drop options

### PR DESCRIPTION
**Pull request by mmitche**
Adds a --channel option, along with shorthands for aspnet, sdk, and runtime repos.
Improve parsing logic too.

### CODE REVIEW

File: src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs [hunk 1]
```diff
@@ -114,56 +114,154 @@ public override async Task<int> ExecuteAsync()
...
+        ///     Returns the repo uri if 
+        /// </summary>
+        /// <returns></returns>
+        private string GetRepoUri()
+        {
+            const string sdkUri = "https://github.com/dotnet/core-sdk";
+            const string runtimeUri = "https://github.com/dotnet/core-setup";
+            const string aspnetUri = "https://github.com/aspnet/AspNetCore";
+            string repoUri = _options.RepoUri;
+            if (string.IsNullOrEmpty(repoUri))
```
Comment (jcagme) [inline src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs]: Nit: add blank lines in between

File: src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs [hunk 2]
```diff
@@ -114,56 +114,154 @@ public override async Task<int> ExecuteAsync()
             }
         }
 
+        /// <summary>
+        ///     Returns the repo uri if
```
Comment (jcagme) [inline src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs]: >Returns the repo uri if  [](start = 16, length = 24)

This seems incomplete

File: src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs [hunk 3]
```diff
@@ -406,24 +504,25 @@ private async Task<DownloadedBuild> GatherDropForBuildAsync(Build build, string
             // If the drop is separated, calculate the directory name based on the last element of the build
             // repo uri plus the build number (to disambiguate overlapping builds)
             string outputDirectory = rootOutputDirectory;
+            string repoUri = build.AzureDevOpsRepository ?? build.GitHubRepository;
```
Comment (jcagme) [inline src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs]: >build.AzureDevOpsRepository ?? build.GitHubRepository; [](start = 29, length = 54)

Is it warrantied that either one of these is set or is there a chance that repoUri could be null/empty? If so, we should add a check so we don't NullRef

Comment (mmitche) [reply src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs]: One will be set.

File: src/Microsoft.DotNet.Darc/src/Darc/Options/GatherDropCommandLineOptions.cs [hunk 4]
```diff
@@ -43,6 +43,18 @@ internal class GatherDropCommandLineOptions : CommandLineOptions
         [Option("include-toolset", HelpText = "Include toolset dependencies.")]
         public bool IncludeToolset { get; set; }
 
+        [Option("sdk", HelpText = "Download an SDK (core-sdk). Requires --commit or --channel.")]
+        public bool DownloadSdk { get; set; }
+
+        [Option("runtime", HelpText = "Download a runtime (core-setup). Requires --commit or --channel.")]
+        public bool DownloadRuntime { get; set; }
+
+        [Option("aspnet", HelpText = "Download an aspnet drop (aspnet) Requires --commit or --channel.")]
```
Comment (jcagme) [inline src/Microsoft.DotNet.Darc/src/Darc/Options/GatherDropCommandLineOptions.cs]: >  [](start = 70, length = 1)

'.' missing

File: src/Microsoft.DotNet.Darc/src/DarcLib/Actions/Remote.cs [hunk 5]
```diff
@@ -383,11 +383,18 @@ public Task<Channel> GetChannelAsync(string channel)
...
-        public Task<Build> GetLatestBuildAsync(string repoUri, int channelId)
+        public async Task<Build> GetLatestBuildAsync(string repoUri, int channelId)
         {
             CheckForValidBarClient();
-            return _barClient.GetLatestBuildAsync(repoUri: repoUri,
-                                                  channelId: channelId);
+            try
+            {
+                return await _barClient.GetLatestBuildAsync(repoUri: repoUri,
+                                                      channelId: channelId);
```
Comment (jcagme) [inline src/Microsoft.DotNet.Darc/src/DarcLib/Actions/Remote.cs]: Nit: this could be a single line or move `repoUri: repoUri` to its own line as well

**Pull request merged**